django/db/models/fields/json.py

```diff
@@ -479,6 +479,29 @@ class KeyTransformGte(KeyTransformNumericLookupMixin, lookups.GreaterThanOrEqual
     pass
 
 
+class KeyTransformIn(lookups.In):
+    def process_rhs(self, compiler, connection):
+        rhs, rhs_params = super().process_rhs(compiler, connection)
+        if not connection.features.has_native_json_field:
+            func = []
+            if connection.vendor == 'oracle':
+                for value in rhs_params:
+                    value = json.loads(value)
+                    function = 'JSON_QUERY' if isinstance(value, (list, dict)) else 'JSON_VALUE'
+                    func.append("%s('%s', '$.value')" % (
+                        function,
+                        json.dumps({'value': value}),
+                    ))
+                rhs_params = []
+            elif connection.vendor == 'mysql' and connection.mysql_is_mariadb:
+                func = ["JSON_UNQUOTE(JSON_EXTRACT(%s, '$'))"] * len(rhs_params)
+            elif connection.vendor in {'sqlite', 'mysql'}:
+                func = ["JSON_EXTRACT(%s, '$')"] * len(rhs_params)
```

Comment:
Really a micro-optmiization, but these can be constructed as tuples and then the `tuple()` cast after the ifs can be moved up to just the oracle branch

```suggestion
                func = ("JSON_UNQUOTE(JSON_EXTRACT(%s, '$'))",) * len(rhs_params)
            elif connection.vendor in {'sqlite', 'mysql'}:
                func = ("JSON_EXTRACT(%s, '$')",) * len(rhs_params)
```